dev/com.ibm.ws.security.jwt/src/com/ibm/ws/security/jwt/internal/ConsumerUtil.java

```diff
@@ -687,5 +695,56 @@ String createDateString(NumericDate date) {
 		// milliseconds
 		return timeUtils.createDateString(1000 * date.getValue());
 	}
+	
+	/**
+	 * Helper method to get the AMR Claim from the jwtClaims.This method checks
+	 * if the value is a string and return singletonList or the ArrayList of
+	 * amrClaims. This is called in validateClaims method
+	 *
+	 */
+	List<String> getJwtAMRList(JwtClaims jwtClaims) throws MalformedClaimException {
+		String claimName = "amr";
+		Object amrObject = jwtClaims.getClaimValue(claimName);
+		if (amrObject instanceof String) {
+			return Collections.singletonList(jwtClaims.getStringClaimValue(claimName));
+		} else if (!(amrObject instanceof List) && amrObject != null) {
+			throw new MalformedClaimException(
+					"The value of the 'amr' claim is not an array of strings or a single string value.");
+		} else {
+			return jwtClaims.getStringListClaimValue(claimName);
+		}
+	}
+
+	/**
+	 * Verifies that values specified in AMR claim is contained in the
+	 * authenticationMethodsReferences list. If allowedAMRClaim is not an array
+	 * then jwtClaims can contain more than required values. If not, then the
+	 * jwtClaimvalues must be a exact match of an element in the array.
+	 */
+	boolean validateAMRClaim(List<String> allowedAmrClaim, List<String> jwtAMRClaims) {
+		boolean valid = false;
+		if (allowedAmrClaim != null && jwtAMRClaims != null) {
+			// If it is not array just check if jwtClaim containsAll and not
+			// equals
+			if (allowedAmrClaim.size() == 1) {
+				List<String> allowedAMRSingle = Arrays.asList(allowedAmrClaim.get(0).split(" "));
+				if (jwtAMRClaims.containsAll(allowedAMRSingle)) {
+					valid = true;
+				}
+			} else {
+				for (String allowedAMR : allowedAmrClaim) {
+					List<String> allowedAMRSingle = Arrays.asList(allowedAMR.split(" "));
+					if (jwtAMRClaims.equals(allowedAMRSingle)) {
+						valid = true;
+						break;
+					}
+
+				}
+			}
+		} else if (allowedAmrClaim == null && (jwtAMRClaims == null || jwtAMRClaims.isEmpty())) {
```

Comment:
This is a default behavior change and cannot be merged in as coded. A previous JWT that included an "amr" claim could have been accepted prior to these changes and it will now be rejected even when a consumer has not been configured with amrValues. Please ensure that a JWT that contains an amr claim is not rejected when amrValues is not configured.

Reply:
I can get rid of this code here and it would take that required part off. @chunlongliang-ibm can confirm if that is what it was intended

Reply:
Fixed it in my new code. Although I have a logic that is if the amrConfig is not specified, then validation returns true. Which means if the amrConfig is specified and the jwt doesnt include amr then the validation will fail. Let me know if I need to add any logic there.